### PR TITLE
[ENOMEM] Return true from _plan_execute only if the buffer is valid

### PR DESCRIPTION
**Pull request by ebraminio**
An alternative fix to https://github.com/harfbuzz/harfbuzz/pull/2606

More details on https://github.com/harfbuzz/harfbuzz/pull/2606#issuecomment-670325875

We are modifying the null pool hosted `hb_buffer_t` (which happens when `hb_buffer_create` gets ENOMEM) by returning true from `hb_shape_plan_execute` which makes `if (res) buffer->content_type = HB_BUFFER_CONTENT_TYPE_GLYPHS;` of `hb_shape_full` be triggered. This changes  returns false on INVALID buffers (which includes null pool hosted `hb_buffer_t`).

### CODE REVIEW

**Comment by ebraminio** [comment]
I think this makes more sense however doing `if (unlikely (hb_object_is_immutable (buffer))) return false;` was idiomatic between similar calls, so not sure about it.

**Comment by ebraminio** [comment]
tl;dr, we are modifying the null pool hosted `hb_buffer_t` by returning true from `hb_shape_plan_execute` which makes `if (res) buffer->content_type = HB_BUFFER_CONTENT_TYPE_GLYPHS;` of `hb_shape_full` be triggered. This returns false on INVALID buffers (which includes null pool hosted `hb_buffer_t`).

**Comment by behdad** [comment]
Pushed alternate fix.
